server/tls/tlscertificate_retriever_file.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Microsoft. All rights reserved.
+
+package tls
+
+import (
+	"crypto"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"golang.org/x/crypto/pkcs12"
+	"io/ioutil"
+)
+
+const (
+	CertLabel = "CERTIFICATE"
+	PrivateKeyLabel = "PRIVATE KEY"
+)
+
+type filetlsCertificateRetriever struct {
+	pemBlock []*pem.Block
+	settings TlsSettings
+}
+
+// GetCertificate Returns the certificate associated with the pfx
+func (fcert *filetlsCertificateRetriever) GetCertificate() (*x509.Certificate, error) {
+	for _, block := range fcert.pemBlock {
+		if block.Type == CertLabel {
+			cert, err := x509.ParseCertificate(block.Bytes)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to parse certificate with error %+v", err)
```

Comment:
should we print the cert name / path for debugging via logs?

Reply:
Name is not requirement in this file. I guess file path maybe useful if someone enters a path to a wrong certificate and it fails to parse but I added a change to include file path but I think its unlikely to be hit.